glusterd2/commands/snapshot/snapshot-create.go

```diff
@@ -378,53 +381,109 @@ func validateSnapCreate(c transaction.TxnCtx) error {
 
 		return errors.New("one or more brick is not compatable")
 	}
-	if nodeData, err = populateBrickMountData(volinfo, req.SnapName); err != nil {
+	if nodeData, err = populateSnapBrickMountData(volinfo, req.SnapName); err != nil {
+
 		return err
 	}
 	c.SetNodeResult(gdctx.MyUUID, snapshot.NodeDataTxnKey, &nodeData)
-	//Quorum check ?
+	//TODO Quorum check has to be implemented once we implement highly available snapshot
 	return nil
 }
+func takeVolumeSnapshots(newVol, oldVol *volume.Volinfo) error {
+	var wg sync.WaitGroup
```

Comment:
Heketi has a nice utility `github.com/heketi/heketi/pkg/utils.StatusGroup` which is a wrapper aroud `sync.Waitgroup` that does exactly what this pattern does. We should maybe create a common utils repo with such utilities and have both heketi and gd2 use them.

Reply:
Hah! I kind of dislike that thing. Not that I dislike the intent, but just the implementation seems wonky and error prone to me.
FWIW some of the heketi utilty code's already been split out into a package here: https://github.com/heketi/utils/blob/master/statusgroup.go However I am not sure what versions the actual heketi code uses as some of the code was coped without having the in tree version removed (IIRC).